chore: update command generator for versioned docs

TerryHowe opened the pull request:
Created a mapping file in case we need to use a bug fix release to generate the docs. I don't think that will happen often.

FeynmanZhou commented on versions.json:
```diff
@@ -1,4 +1,4 @@
 [
-  "1.2",
-  "1.1"
+  "1.2.0",
```
If we specify the patch version number here, it will require us update the version number per every patch release. However, we probably not need to do this since 1.2.1 and 1.2.0 docs shouldn't differ according to SemVer because there are no new features in a new patch release. 

I would suggest keeping it as minor version or use `1.1.x`, `1.2.x`.

TerryHowe replied:
Done

FeynmanZhou reviewed (commented):
Please see my comments above

TerryHowe commented:
@FeynmanZhou I removed the bug fix release from the script and added a way to map using bug fixes if necessary. Normally, the pages won't change during a release.

qweeah reviewed (approved):
~~LGTM although IANAM.~~ What if we are updating 1.1 and the content of version-latest is 2.1.1?

Better add some comments or documents on how `versions-latest` should be used, otherwise when someone need to refresh for a patch release, they need to read and understand below code in `refresh-commands.sh`
```bash
map_version() {
    grep "$1" versions-latest || echo "$1.0"
}
```

qweeah commented on tools/refresh-commands.sh:
```diff
@@ -35,20 +33,31 @@ list_commands() {
-WEIGHT=10
-list_commands | while read COMMAND
+map_version() {
+    grep "$1" versions-latest || echo "$1.0"
```
What if we are updating 1.1 and the content of version-latest is `2.1.1`?

qweeah commented:
LGTM although IANAM.

FeynmanZhou commented on tools/refresh-commands.sh:
```diff
@@ -2,12 +2,28 @@
+# use `1.2.2` to generate the documentation, put a line in the `versions-latest`
+# file `1.2.2`.  If there is no line in the `versions-latest` that matches
+# the major and minor release `1.2`, then `1.2.0` will be used to generate
+# the documentation.
```
@TerryHowe It seems the generator script can be tested by maintainers after this PRs merged into `main`, right?

TerryHowe replied:
Yes, it will run with the `npm build`, but right now, the `npm build` does not fail if this fails.

TerryHowe replied:
As soon as this merges, I'll create a PR with the latest updates.

FeynmanZhou reviewed (approved):
LGTM. Thanks @TerryHowe
